compiler/backend/src/org/jetbrains/kotlin/codegen/inline/AnonymousObjectTransformer.kt

```diff
@@ -113,22 +114,12 @@ class AnonymousObjectTransformer(
             override fun visitEnd() {}
         }, ClassReader.SKIP_FRAMES)
 
-        if (!inliningContext.isInliningLambda) {
-            sourceMapper = if (debugInfo != null && !debugInfo!!.isEmpty()) {
-                SourceMapper.createFromSmap(SMAPParser.parse(debugInfo!!))
-            } else {
-                //seems we can't do any clever mapping cause we don't know any about original class name
-                IdenticalSourceMapper
-            }
-            if (sourceInfo != null && !GENERATE_SMAP) {
-                classBuilder.visitSource(sourceInfo!!, debugInfo)
-            }
-        } else {
-            if (sourceInfo != null) {
-                classBuilder.visitSource(sourceInfo!!, debugInfo)
-            }
-            sourceMapper = IdenticalSourceMapper
-        }
+        // When regenerating objects in lambdas, keep the old SMAP and don't remap the line numbers to save time.
+        // The result is effectively the same anyway.
```

Comment:
If object inlines smth inside itself will this mapping be lost with this optimization?

Reply:
What do you mean? There is nothing to inline into objects inside lambdas at this point - the purpose of regenerating them is only to replace `this$0`.

Reply:
I meant  next case:
```
package test

inline fun test(s: () -> Unit) {
    s()
}

inline fun test2(crossinline s: () -> Unit) {
    s()
}
```
```
import test.test
import test.test2


fun box(): String {
    var z = "fail"

    test2 {
        { // regenerated object in inline lambda
            test { //will mapping for this call is lost?
                z = "OK"
            }
        }()
    }
    return z;
}
```

Reply:
No, the ranges for this call are already present in the original object and will be copied to the regenerated version.